### PR TITLE
`Cp2kBaseWorkChain`: Fix restart for no `output_structure`

### PR DESCRIPTION
**Pull request by mbercx**
For some CP2K calculations, e.g. a "single-point" calculation, there is
no `output_structure` output in the `Cp2kCalculation`. In this case the
current `Cp2KBaseWorkChain` fails to restart, since the
`overwrite_input_structure` step in the outline _expects_ this output to
be present.

Here we simply add a check to see if `output_structure` is one of the
outputs of the final `Cp2kCalculation`, and only update the input
structure in the context if that is the case.

### CODE REVIEW

**Comment by mbercx** [comment]
@yakutovicha this fix is necessary so the tests in https://github.com/lsmo-epfl/aiida-lsmo/pull/98 complete successfully.

**Comment by mbercx** [comment]
Hmm, seems one example test fails because it's expecting the example that runs into the wall time to still have an `output_structure`:

https://github.com/aiidateam/aiida-cp2k/blob/59a795d08de6a8e20eaa7494147b2ea4192741ae/examples/single_calculations/example_restart.py#L126-L134

Strange that this test fails now though, since my changes were only to the `Cp2kBaseWorkChain`, and I believe this test just runs a manual restart of the `Cp2kCalculation`.

Also pinging @mpougin and @VasylStotskyi for comments.

**Comment by mpougin** [comment]
thank you for your help @mbercx, running singlepoint calculations works fine now

**Comment by yakutovicha** [comment]
Hey @mbercx 👋. I am on the PR now. The change looks good to me, but the failure of the test is indeed confusing. I will try to rerun it one more time (maybe it helps). In case the issue persists - I will have a closer look.

**Comment by yakutovicha** [comment]
hehe, as expected tests are now passing.

**Review by yakutovicha — APPROVED** [review]
LGTM!

**Comment by mpougin** [comment]
thanks for your help @yakutovicha and @mbercx :)

**Comment by mpougin** [comment]
now the tests on aiida-ddec for singlepoint calculations still fail on github, though they run locally when I am on the cp2k develop branch. I assume aiida-ddec uses the latest release? @yakutovicha can you tell me how to fix this?

**Comment by yakutovicha** [comment]
> @yakutovicha can you tell me how to fix this?

@mpougin sorry, I do not intend to work on `aiida-ddec`, as I've already mentioned. Telling how to fix the issue would require looking at the package and trying to understand where the error is coming from.

My guess would be to wait for the next release of `aiida-cp2k`. This will happen right after I got approval from @mbercx on the  #152. It might be able to fix it.

**Comment by mpougin** [comment]
Yes, I got that message. 
I know that the error comes from this issue and is fixed by @mbercx commit. So you answered my question which was just a technicality, thanks, I just have to wait for the release. I thought it would take longer

**Pull request merged**
